helm/cluster-azure/templates/_helpers.tpl

```diff
@@ -46,12 +46,18 @@ room for such suffix.
 {{- end -}}
 
 {{/*
-The default name for our images in the Community Gallery is  "capi-flatcar-stable-<KUBERNETES_VERSION>-gen2-gs",
+The default name for our images in the Community Gallery is  "capi-flatcar-stable-<KUBERNETES_VERSION>-gen2-<IMAGE_VARIANT>-gs",
 use it when no value is passed in
 */}}
 {{- define "flatcarImageName" -}}
```

Comment:
@tuladhar Related to yesterday's SIG architecture talk and the change here.

IIUC we want to use image variant for now, so we don't overwrite OS images, and that's all good.

OTOH we should not have OS image name (nor name parts) configurable. Maybe it's fine to have it in internal like here for the sake of testing, but changes of OS images should always be done via cluster-\<provider\> PRs that change `capi-flatcar-stable-<k8s version>-gen2-<variant>` version string, and then clusters are upgraded to a newer cluster-\<provider\> version. So we should never set `.Values.internal.kubernetesVersion` nor `.Values.internal.image.*` in e.g. gitops repos.

e.g. In cluster chart for overriding app versions for example (for development/testing/ops purposes), we are putting app version overrides into a new `.Values.internal.ephemeralConfiguration` property with a big disclaimer about how, when and why to use it and that it must be only used for some temporary purpose (there is no mechanism that would remove/overwrite those `.Values.internal.ephemeralConfiguration` values, but there might be in the future 😄 😉 

No changes are required here in this PR now, but more of a heads up that we (Turtles) might change how and where OS image is configured for temporary development/testing/ops purposes (e.g. to use `.Values.internal.ephemeralConfiguration`), and that you should not rely on OS image being configurable.